step-01/js/main.js

```diff
@@ -1,26 +1,73 @@
 'use strict';
 
-navigator.getUserMedia = navigator.getUserMedia ||
-    navigator.webkitGetUserMedia || navigator.mozGetUserMedia;
+// Define navigator.mediaDevices.getUserMedia() for older devices.
 
-var constraints = {
+// Old browsers might not implement mediaDevices, so we set an empty object.
+if (navigator.mediaDevices === undefined) navigator.mediaDevices = {};
+
+// Add mediaDevices.getUserMedia method if it's missing.
+if (navigator.mediaDevices.getUserMedia === undefined) {
+
+  /**
+   * Gets a stream of media content containing tracks of video or audio.
+   * Produces a Promise which resolves into a MediaStream object.
+   * @param {!MediaStreamConstraints} constraints to apply to the media stream.
```

Comment:
Nothing against JSDoc, but we don't use it anywhere else in the codelabs, so probably shouldn't use here.

Comments are fine.

Reply:
Removed JSDoc.